components/page-templates/B2B/QuickOrderTemplate/QuickOrderTemplate.tsx

```diff
@@ -0,0 +1,205 @@
+              </Button>
+            </Stack>
+          ) : null}
+        </Grid>
+        <Grid item xs={12} md={4}>
+          <B2BProductSearch onAddProduct={handleAddProduct} />
+        </Grid>
+        <Grid item xs={12}>
+          <Stack gap={3}>
+            {tabAndDesktopScreen ? (
+              <QuickOrderTable
+                cartItems={cartItems as CrCartItem[]}
+                fulfillmentLocations={fulfillmentLocations}
+                purchaseLocation={purchaseLocation}
+                onFulfillmentOptionChange={onFulfillmentOptionChange}
+                onQuantityUpdate={handleQuantityUpdate}
+                onStoreSetOrUpdate={handleProductPickupLocation}
+              />
+            ) : (
+              <Stack spacing={2}>
+                <Typography variant="h2">{t('cart')}</Typography>
+                {cartItems.length > 0 ? (
+                  <CartItemList
+                    cartItems={cartItems}
+                    fulfillmentLocations={fulfillmentLocations as Location[]}
+                    purchaseLocation={purchaseLocation}
+                    onCartItemDelete={() => null}
+                    onCartItemQuantityUpdate={handleQuantityUpdate}
+                    onFulfillmentOptionChange={onFulfillmentOptionChange}
+                    onProductPickupLocation={handleProductPickupLocation}
+                    onCartItemActionSelection={() => null}
+                  />
+                ) : (
+                  <Typography variant="body1" sx={quickOrderTemplateStyles.noCartItems}>
+                    {t('search-to-add-products')}
+                  </Typography>
+                )}
+              </Stack>
+            )}
+
+            {!tabAndDesktopScreen && cartItems.length ? (
+              <Stack spacing={2}>
+                <Button variant="contained" color="primary">
+                  {t('checkout')}
+                </Button>
+                <Button variant="contained" color="secondary">
+                  {t('initiate-quote')}
+                </Button>
+              </Stack>
+            ) : null}
+          </Stack>
+        </Grid>
+        {tabAndDesktopScreen ? (
+          <Grid item xs={12}>
+            <Stack sx={quickOrderTemplateStyles.promoCode}>
+              <PromoCodeBadge
+                onApplyCouponCode={handleApplyCouponCode}
+                onRemoveCouponCode={handleRemoveCouponCode}
+                promoError={!!promoError}
+                helpText={promoError}
+                couponLabel="Coupon"
+              />
+              <KeyValueDisplay
+                option={{
+                  name: t('order-total'),
+                  value: `${t('currency', { val: cartTotal })} `,
+                }}
+                variant="body1"
+                sx={quickOrderTemplateStyles.orderTotal}
+              />
+            </Stack>
+          </Grid>
+        ) : null}
+      </Grid>
+    </>
+  )
+}
+
+export default QuickOrderTemplate
```

Comment:
Group logical code
Instead of if (!productGetters.isVariationProduct(product)) else {}
use (productGetters.isVariationProduct(product)) else {}

I think till now we are considering tab in mobile only (and not with desktop)
const tabAndDesktopScreen = useMediaQuery((theme: Theme) => theme.breakpoints.up('sm'))

Reply:
Resolved